presto-spi/src/main/java/com/facebook/presto/spi/connector/ConnectorMetadata.java

```diff
@@ -130,6 +131,14 @@ default ConnectorTableLayoutHandle getAlternativeLayoutHandle(ConnectorSession s
         return Optional.empty();
     }
 
+    /**
+     * Provides partitioning handle for connector partitioning
+     */
+    default ConnectorPartitioningHandle getPartitioningHandle(ConnectorSession session, int partitionCount, List<Type> partitionTypes)
```

Comment:
Should the name be `getPartitioningHandleForExchange`? -- The rational is connector, in theory, can have multiple partitioning mechanisms when data is stored. But there is one "preferred" for exchange. The name also make it a bit clearer why we need this method (or you may not agree :) ) 

I understand `getPartitioningHandleForExchange` has its own problems... So I don't have strong opinions here.

Reply:
Actually yes, i like `getPartitioningHandleForExchange` more